### DISCUSSION TITLE
서버에서 반환하는 오류메시지가 자세하다는 생각이 들어요

### DISCUSSION THREAD

**Discussion by WaiNaat** [woowacourse-teams/2023-pium#341] [뜨거운 토론]
현재 프로덕션 서버의 상황입니다.

1. 로그인한 사용자가 타인의 반려 식물에 접근했을 때
```
400 bad request
{"message":"요청 사용자와 반려 식물의 사용자가 일치하지 않습니다. memberId: {number}"}
```
2. 로그인한 사용자가 존재하지 않는 반려 식물에 접근했을 때
```
404 not found
{"message":"일치하는 반려 식물이 존재하지 않습니다. id :{number}"}
```

다행히 저희 서비스는 민감한 개인 정보를 담고 있지 않지만 사용자에게 자료가 없는 것과 있지만 접근할 수 없다는 걸 알려주는 게 위험해 보이는데 어떻게 생각하시나요??

**Comment by Choi-JJunho** [comment]
현재 프론트엔드에서 서버에서 반환되는 오류메시지를 이용해서 특별한 처리를 하고있지않다면 오류메시지를 보다 추상적으로 구성해도 좋을 것 같네요

ex) 
404 - 자원을 찾을 수 없습니다
400 - 잘못된 요청입니다
401 - 잘못된 인증정보입니다

500 - 서버에서 오류가 발생했습니다

**Comment by Kim0914** [comment]
현재 저희가 작성한 예외 메세지는 사용자가 아닌 클라이언트를 위한 예외 메시지를 전달하고 있습니다 !

이 예외 메세지들이 웹 브라우저 개발자 도구에서 확인할 수 있기 때문에 말씀해주신건지 궁금합니다 ㅎㅎ

프론트엔드에서 서버에서 반환하는 상태 코드를 기반으로 사용자에게 보여지는 메세지를 추상화 할 수 있나요?!

**Reply by WaiNaat** [reply]
상태 코드만 주셔도 프론트에서 그거랑 어떤 문자열이랑 연결짓는 방식으로 메시지를 전달할 순 있어요
제가 질문했던 이유는 개발자 도구에서 상태 코드를 보고 해당 자료의 유무에 대한 정보를 쉽게 얻을 수 있어서 그걸 숨길 수 있는 방법이 있나 궁금했어서였어요 단순히 오류 메시지 문제가 아니라 상태 코드 자체도 정보가 되니까요
근데 생각해보면 마땅한 방법이 없는 것 같기도 합니다

**Comment by rawfishthelgh** [comment]
사용자와 관련한 예외일 경우 추상화시키는게 좋을 듯 하네요
ex) 요청 사용자와 반려 식물의 사용자가 일치하지 않습니다 -> 잘못된 접근입니다
